qiskit_optimization/algorithms/optimization_algorithm.py

```diff
@@ -99,7 +99,7 @@ class OptimizationResult:
     def __init__(
         self,
         x: Optional[Union[List[float], np.ndarray]],
-        fval: float,
+        fval: Optional[float],
```

Comment:
If fval is passed in as None I imagine that is how it remains - so the fval getter now should be updated for its return typehint o say Optional[float] too right?

Reply:
If `x` is `None`, `fval` is needed to be `None` as well. It happens in this line. https://github.com/Qiskit/qiskit-optimization/blob/7f341b00d850577cf866caecbee61a102144156e/qiskit_optimization/algorithms/minimum_eigen_optimizer.py#L235-L238
But, it perhaps won't happen according to https://github.com/Qiskit/qiskit-optimization/blob/7f341b00d850577cf866caecbee61a102144156e/qiskit_optimization/algorithms/optimization_algorithm.py#L501
Let me check more details.

Reply:
`eigen_result.eigenstate` can be `None` because eigenstate is `Optional[ndarray]`. So, `fval` needs to be `None` in case `eigenstate` is `None`.
https://github.com/Qiskit/qiskit-optimization/blob/7f341b00d850577cf866caecbee61a102144156e/qiskit_optimization/algorithms/minimum_eigen_optimizer.py#L218
```python
class MinimumEigensolverResult(AlgorithmResult):
    """Minimum Eigensolver Result."""
...
    @property
    def eigenstate(self) -> Optional[np.ndarray]:
        """return eigen state"""
        return self._eigenstate
```
https://github.com/Qiskit/qiskit-terra/blob/0167c156d2e2c929fd9041ef75762905ae826ccc/qiskit/algorithms/minimum_eigen_solvers/minimum_eigen_solver.py#L87

Reply:
I updated the type of `OptimizationResult.fval` to be `Optional[float]`.